osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs

```diff
@@ -282,13 +296,84 @@ public override Task ChangeUserMods(IEnumerable<APIMod> newMods)
             return Task.CompletedTask;
         }
 
+        private CancellationTokenSource? countdownFinishSource;
+        private CancellationTokenSource? countdownStopSource;
+        private Task countdownTask = Task.CompletedTask;
+
+        public void FinishCountDown() => countdownFinishSource?.Cancel();
+
         public override async Task SendMatchRequest(MatchUserRequest request)
         {
             Debug.Assert(Room != null);
             Debug.Assert(LocalUser != null);
 
             switch (request)
             {
+                case StartMatchCountdownRequest matchCountdownRequest:
+                    countdownStopSource?.Cancel();
+
+                    var stopSource = countdownStopSource = new CancellationTokenSource();
+                    var finishSource = countdownFinishSource = new CancellationTokenSource();
+                    var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(stopSource.Token, finishSource.Token);
+                    var countdown = new MatchStartCountdown { EndTime = DateTimeOffset.Now + matchCountdownRequest.Delay };
+
+                    Task lastCountdownTask = countdownTask;
+                    countdownTask = start();
+
+                    async Task start()
+                    {
+                        try
+                        {
+                            await lastCountdownTask;
+                        }
+                        catch (OperationCanceledException)
+                        {
+                        }
```

Comment:
What's this for?

Reply:
Was copied from the osu-server-spectator implementation, but is unnecessary here (and the server implementation has changed to catch all exceptions instead anyway).